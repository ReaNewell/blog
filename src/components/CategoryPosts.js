import React from 'react';
import SummarizedPost from './SummarizedPost';

class CategoryPosts extends React.Component {
    render() {
        return (
            <div className="recent-posts">
                <h2 className="recent-posts__title">{`Articles Related To ${this.props.category}`}</h2>
                {this.props.posts.map((post) => (
                    <SummarizedPost 
                        key = {post.id}
                        title = {post.title}
                        imageName = {post.postPictureName}
                        imagePath = {post.postPicture}
                        date = {post.postDate}
                    />
                ))}
            </div>
        )
    }
}

export default CategoryPosts;