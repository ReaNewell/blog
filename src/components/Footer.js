import React from 'react';

export const Footer = () => (
    <div className="footer">
        <img alt='site-logo' src='/images/rea-logo.png'/>
    </div>
);