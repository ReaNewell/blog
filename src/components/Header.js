import React from 'react';
import { Navigation } from './Navigation';

export const Header = () => (
    <div>
        <Navigation />
    </div>
);