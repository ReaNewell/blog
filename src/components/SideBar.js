import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import categoriesSelector from '../selectors/categories';

export class SideBar extends React.Component {
    render() {
        return (
            <div className="sidebar">
                <h3 className="sidebar__heading">Find me online!</h3>
                <a className="social-media" href="https://www.reanewell.com" target="_blank"><img src="/images/portfolio.png" alt="My Portfolio"/></a>
                <a className="social-media" href="https://instagram.com/rea.webdev" target="_blank"><img src="/images/instagram.jpg" alt="instagram logo"/></a>
                <a className="social-media" href="https://codepen.io/reanewell/" target="_blank"><img src="/images/codepen.png" alt="codepen logo"/></a>
                <a className="social-media" href="https://twitter.com/ReaNewell" target="_blank"><img src="/images/twitter.jpg" alt="twitter logo"/></a>
                <hr />
                <h3 className='sidebar__heading--green'>Categories</h3>
                { this.props.categories.map((category, index) => (
                    <Link
                        className='sidebar__category'
                        key={index}
                        to={`/categories/${category}`}
                    >{category}</Link>
                ))}
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    categories: categoriesSelector(state.posts)
})

export default connect(mapStateToProps)(SideBar);