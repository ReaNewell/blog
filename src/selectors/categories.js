export default (posts) => {
    let categories = [];
    posts.forEach(post => {
        post.categories.forEach(category => {
            if (!categories.includes(category)) {
                categories.push(category);
            }
        })
    })
    return categories;
};