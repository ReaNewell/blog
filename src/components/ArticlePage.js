import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { Footer } from './Footer';
import { Header } from './Header';
import SummarizedPost from './SummarizedPost';
import SideBar from './SideBar';
import moreArticles from '../selectors/moreArticles';
import dateFormatter from '../tools/dateFormatter';

export class ArticlePage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            html: this.props.post.body.replace(/^"(.*)"$/, '$1')
        }
    } 
    componentDidMount() {
        window.scroll(0, 0);

        // Selects this the body element of this component and assign the HTML value of the post to that element.
        const body = ReactDOM.findDOMNode(this).getElementsByClassName('article__body')[0];
        body.innerHTML = this.state.html;
    }
    componentDidUpdate() {
        window.scroll(0, 0);
    }
    render() {
        return (
            <div className="article-page">
                <Header />
                    <article className="article">
                        <h2 className="article__title">{this.props.post.title.toUpperCase()}</h2>
                        <p className="article__date">{dateFormatter(this.props.post.postDate.day, this.props.post.postDate.month, this.props.post.postDate.year)}</p>
                        {this.props.post.postPicture && <img className="article__image" src={this.props.post.postPicture} alt={this.props.post.postPictureName}/>}
                        <div className="article__body" />

                        <div className='share-buttons'>
                            <a className="share-buttons__share-button" href={`http://facebook.com/sharer.php?u=https%3A%2F%2Fblog.reanewell.com/posts/${this.props.post.title.replace(/ /g, "-")}`}>Facebook</a>
                            <a className="share-buttons__share-button" href={`https://twitter.com/intent/tweet?url=https%3A%2F%2Fblog.reanewell.com/${this.props.post.title.replace(/ /g, "-")}`}>Twitter</a>
                            <a className="share-buttons__share-button" href={`https://www.linkedin.com/shareArticle?mini=true&url=https%3A%2F%2Fblog.reanewell.com/${this.props.post.title.replace(/ /g, "-")}&title=${this.props.post.title}`}>LinkedIn</a>
                        </div>

                        <h2 className='article__header'>More Articles</h2>
                        {this.props.moreArticles.map((post) => (
                            <SummarizedPost 
                                key = {post.id}
                                title = {post.title}
                                imageName = {post.postPictureName}
                                imagePath = {post.postPicture}
                                date = {post.postDate}
                            />
                        ))}
                    </article>
                <SideBar />
                <Footer />
            </div>
        )
    }
}
const mapStateToProps = (state, props) => {
    const post = state.posts.find((post) => post.title === props.match.params.title.replace(/-/g, ' '));

    return ({
        post,
        moreArticles: moreArticles(state.posts, post.id, post.category)
    })
};

export default connect(mapStateToProps)(ArticlePage);