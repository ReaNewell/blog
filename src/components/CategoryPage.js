import React from 'react';
import { connect } from 'react-redux';
import categoryFilter from '../selectors/categoryFilter';
import { Footer } from './Footer';
import { Header } from './Header';

import SideBar from './SideBar';
import CategoryPosts from './CategoryPosts';

export const CategoryPage = (props) => {
    componentDidMount = () => {
        window.scroll(0, 0);
    }
    componentDidUpdate = () => {
        window.scroll(0, 0);
    }
    return(
        <div className="homepage">
            <Header />
            <CategoryPosts 
                category={props.category}
                posts={props.posts}
            />
            <SideBar />
            <Footer />
        </div>
    )
};
const mapStateToProps = (state, props) => ({
    category: props.match.params.category,
    posts: categoryFilter(state.posts, props.match.params.category)
});

export default connect(mapStateToProps)(CategoryPage);