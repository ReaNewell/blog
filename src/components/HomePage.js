import React from 'react';

import { Footer } from './Footer';
import { Header } from './Header';
import RecentPosts from './RecentPosts';
import SideBar from './SideBar';

const HomePage = () => {
    window.scroll(0, 0);
    return (
        <div className="homepage">
            <Header />
            <RecentPosts />
            <SideBar />
            <Footer />
        </div>
    )
};

export default HomePage;