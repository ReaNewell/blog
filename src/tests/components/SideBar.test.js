import React from 'react';
import { shallow } from 'enzyme';
import { SideBar } from '../../components/SideBar';
import categories from '../fixtures/categories';

test('should render SideBar component correctly', () => {
    const wrapper = shallow(<SideBar categories={categories}/>);
    expect(wrapper).toMatchSnapshot();
});