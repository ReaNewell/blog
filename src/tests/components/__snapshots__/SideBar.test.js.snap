// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SideBar component correctly 1`] = `
<div
  className="sidebar"
>
  <h3
    className="sidebar__heading"
  >
    Find me online!
  </h3>
  <a
    className="social-media"
    href="https://www.reanewell.com"
    target="_blank"
  >
    <img
      alt="My Portfolio"
      src="/images/portfolio.png"
    />
  </a>
  <a
    className="social-media"
    href="https://instagram.com/rea.webdev"
    target="_blank"
  >
    <img
      alt="instagram logo"
      src="/images/instagram.jpg"
    />
  </a>
  <a
    className="social-media"
    href="https://codepen.io/reanewell/"
    target="_blank"
  >
    <img
      alt="codepen logo"
      src="/images/codepen.png"
    />
  </a>
  <a
    className="social-media"
    href="https://twitter.com/ReaNewell"
    target="_blank"
  >
    <img
      alt="twitter logo"
      src="/images/twitter.jpg"
    />
  </a>
  <hr />
  <h3
    className="sidebar__heading--green"
  >
    Categories
  </h3>
  <Link
    className="sidebar__category"
    key="0"
    replace={false}
    to="/categories/html"
  >
    html
  </Link>
  <Link
    className="sidebar__category"
    key="1"
    replace={false}
    to="/categories/css"
  >
    css
  </Link>
  <Link
    className="sidebar__category"
    key="2"
    replace={false}
    to="/categories/javascript"
  >
    javascript
  </Link>
</div>
`;
